contractcourt/channel_arbitrator_test.go

```diff
@@ -58,6 +58,34 @@ func (b *mockArbitratorLog) FetchUnresolvedContracts() ([]ContractResolver,
 	for resolver := range b.resolvers {
 		v[idx] = resolver
 		idx++
+
+		// In order to permit restarts, for all possible resolvers,
+		// we'll give them a fresh quit channel. This is required as
+		// otherwise, if a restart is simulated within a test (stopping
+		// the existing channel arb along the way), then the resolvers
+		// will immediately exit, as their quit channel is still
+		// closed.
```

Comment:
Especially because we want to test the contract resolution executes properly after a restart, I don't think we should reuse the resolver instances and their internal state. This also relies on knowledge of the internal implementation of the resolvers, which makes for brittle tests.

I think that creating a temp db and using the real arbitrator log is a better option.

Reply:
setting the quit channels in the below cases also produces a race condition in the unit tests, perhaps new resolvers are better as joost suggested

Reply:
Modified to use a real DB, added the _start_ of a test context along the way. There's still some weirdness since the arb log inherits part of the config of the channel arb that initially creates it, but this'll require a bigger overhual to rectify.